docs/pages/guides/keyboard-handling.mdx

```diff
@@ -0,0 +1,262 @@
+        <Stack.Screen name="advanced" />
+        <Stack.Screen name="+not-found" />
+      </Stack>
+    </KeyboardProvider>
+  );
+}
+```
+
+### Usage
+
+Let's explore how to use the `useKeyboardHandler` hook to achieve the same functionality that `KeyboardAvoidingView` provides.
+
+The `useKeyboardHandler` hook takes two parameters: a handler and dependencies. The handler provides information about the current device's keyboard. You'll use the `onMove` handler to get the keyboard height and use it to push the screen's content above the keyboard when an input is active:
+
+```tsx ChatScreen.tsx
+import { useKeyboardHandler } from 'react-native-keyboard-controller';
+import Animated, { useAnimatedStyle, useSharedValue } from 'react-native-reanimated';
+
+const useGradualAnimation = () => {
+  const height = useSharedValue(0);
+
+  useKeyboardHandler(
+    {
+      onMove: e => {
+        'worklet';
+        height.value = Math.max(e.height, 0);
+      },
+    },
+    []
+  );
+  return { height };
+};
+```
+
+The above example uses the `useSharedValue` hook from Reanimated to store the keyboard's height. The value changes from 0 when the keyboard is dismissed to the maximum height when it's active.
+
+Now, let's use the `useGradualAnimation` hook inside a screen component:
+
+```tsx ChatScreen.tsx
+import { StyleSheet, Platform, FlatList, View, StatusBar, TextInput } from "react-native";
+import Animated, { useAnimatedStyle, useSharedValue } from "react-native-reanimated";
+import { useKeyboardHandler } from "react-native-keyboard-controller";
+
+import MessageItem from "@/components/MessageItem";
+import { messages } from "@/messages";
+
+const useGradualAnimation = () => {...};
+
+export default function ChatScreen() {
+  const { height } = useGradualAnimation();
+
+  const fakeView = useAnimatedStyle(() => {
+    return {
+      height: Math.abs(height.value),
+    };
+  }, []);
+
+  return (
+    <View style={styles.container}>
+      <FlatList
+        data={messages}
+        renderItem={({ item }) => <MessageItem message={item} />}
+        keyExtractor={item => item.createdAt.toString()}
+        contentContainerStyle={styles.listStyle}
+      />
+      <TextInput placeholder="Type a message..." style={styles.textInput} />
+      <Animated.View style={fakeView} />
+    </View>
+  );
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    paddingTop: Platform.OS === "android" ? StatusBar.currentHeight : 0,
+  },
+  listStyle: {
+    padding: 16,
+    gap: 16,
+  },
+  textInput: {
+    width: "95%",
+    height: 45,
+    borderWidth: 1,
+    borderRadius: 8,
+    borderColor: "#d8d8d8",
+    backgroundColor: "#fff",
+    padding: 8,
+    alignSelf: "center",
+    marginBottom: 8,
+  },
+});
+```
+
+The above uses the keyboard's height to move the screen content above the keyboard. The view after the keyboard has an animated style that modifies its height based on the keyboard's height.
+
+### Handling multiple inputs
```

Comment:
This feels like the more basic/essential scenario than smoothly animating a toolbar input, maybe could go above that.

Reply:
You're right! Although this is also the last part of the video. Do you think having this part first in the guide and last in the video will be confusing? Maybe we can leave it as a bonus part?

Reply:
I think it's OK if the video is in a different sequence. It's more like showing through a practical example than summarizing the different techniques.